db/worker_lifecycle.go

```diff
@@ -24,6 +25,26 @@ func NewWorkerLifecycle(conn Conn) WorkerLifecycle {
 	}
 }
 
+func (lifecycle *workerLifecycle) DeleteUnresponsiveEphemeralWorkers() ([]string, error) {
+	query, args, err := psql.Delete("workers").
+		Where(sq.Eq{"ephemeral": true}).
+		Where(sq.Expr("expires < NOW()")).
+		Where(sq.Eq{"state": string(WorkerStateRunning)}).
```

Comment:
Hmm could this result in a situation where you've started landing/retiring an ephemeral worker but then it never goes away when it's unresponsive?

Should we just remove the state check?